###author, forum, title, text, 여태까지 한 것들을 불러와서 하나로 합침

In [26]:
import pandas as pd
import numpy as np
import json
import pickle

num_author = 1000
num_title = 1000
num_text = 1000

author_forum_p = pickle.load(open("author, forum_%d.p" % num_author, "rb"))
title_p = pickle.load(open("title_%d.p" % num_title, "rb"))
text_p = pickle.load(open("text_%d.p" % num_text, "rb"))

train = pd.concat([author_forum_p, title_p, text_p], axis = 1)
#train = pd.concat([title_p, text_p], axis = 1)

#train = author_forum_p

### predictor, model 생성

#### Random Forest

In [35]:
import xgboost as xgb
from sklearn.ensemble import RandomForestClassifier

label = 'istroll'
pre = train.columns.drop([label, "forumid"])

model = RandomForestClassifier(n_estimators=10, n_jobs=6)

#### XGBoost

In [ ]:
from xgboost.sklearn import XGBClassifier

# label = 'istroll'
label = '3'
labeled_train.columns = [str(x) for x in range(len(labeled_train.columns))]
pre = labeled_train.columns.drop(['0', '1', '3'])
# pre = ['author', 'forumid']

model = XGBClassifier()

###cross validation

In [36]:
from sklearn import cross_validation

cv_value = 3

scores = cross_validation.cross_val_score(model, train[pre], train[label], cv=cv_value, scoring="roc_auc")
cv_result = scores.mean()

print(cv_result)

0.749510340022


##Result

###RF
####morphs
   * forumid와 author, title, text를 bag of words로 변환 (feature : 1,000개) - 0.721
   * forumid와 author, title을 bag of words로 변환 (feature : 1,000개) - 0.712577335639
   * forumid와 author, text를 bag of words로 변환 (feature : 1,000개) - 0.721
   
####pos
   * forumid와 author, title, text의 pos를 bag of words로 변환 (feature : 1,000개) - 0.728390543803
   * forumid와 author, title, text의 pos를 LDA로 변환 (keep_n : 5,000, num_topics: 1,000) - 0.662467774802
   * forumid와 author, title의 pos를 bag of words로 변환 (feature : 1,000개) - 0.704123734689
   * forumid와 author, text의 pos를 bag of words로 변환 (feature : 1,000개) - 0.720707529009
   * forumid와 author, title, text의 pos를 word2vec으로 변환 (feature : 300개) - 0.729942013224
   * forumid와 author, text의 pos를 word2vec으로 변환 (feature : 300개) - 0.749510340022
   
###XGBoost
####pos
   * forumid와 author, title, text의 pos를 bag of words로 변환 (feature : 1,000개) - 0.662807681025